projects/storefrontlib/src/lib/store-finder/services/store-finder.service.ts

```diff
@@ -8,43 +8,57 @@ import * as fromStore from '../store';
 
 @Injectable()
 export class StoreFinderService {
+  private geolocationWatchId: number;
+
   constructor(
     private store: Store<fromStore.StoresState>,
     private winRef: WindowRef
   ) {}
 
   findStores(queryText: string, useMyLocation?: boolean) {
     if (useMyLocation) {
-      this.store.dispatch(new fromStore.OnHold());
-      this.winRef.nativeWindow.navigator.geolocation.getCurrentPosition(
+      this.clearWatchGeolocationAndDispatchAction(new fromStore.OnHold());
+      this.geolocationWatchId = this.winRef.nativeWindow.navigator.geolocation.watchPosition(
         (pos: Position) => {
           const longitudeLatitude: LongitudeLatitude = {
             longitude: pos.coords.longitude,
             latitude: pos.coords.latitude
           };
-          this.store.dispatch(
+          this.clearWatchGeolocationAndDispatchAction(
             new fromStore.FindStores({ queryText, longitudeLatitude })
           );
         }
       );
     } else {
-      this.store.dispatch(new fromStore.FindStores({ queryText }));
+      this.clearWatchGeolocationAndDispatchAction(
+        new fromStore.FindStores({ queryText })
+      );
     }
   }
 
   viewAllStores() {
-    this.store.dispatch(new fromStore.ViewAllStores());
+    this.clearWatchGeolocationAndDispatchAction(new fromStore.ViewAllStores());
   }
 
   viewAllStoresForCountry(countryIsoCode: string) {
-    this.store.dispatch(
+    this.clearWatchGeolocationAndDispatchAction(
       new fromStore.FindAllStoresByCountry({ countryIsoCode })
     );
   }
 
   viewAllStoresForRegion(countryIsoCode: string, regionIsoCode: string) {
-    this.store.dispatch(
+    this.clearWatchGeolocationAndDispatchAction(
       new fromStore.FindAllStoresByRegion({ countryIsoCode, regionIsoCode })
     );
   }
+
+  private clearWatchGeolocationAndDispatchAction(action: Action) {
+    if (this.geolocationWatchId != null) {
```

Comment:
!==

Reply:
make the initial value of watchId null and no need to deal with undefined. and you can use !==